cmd/libsnap-confine-private/utils.c

```diff
@@ -261,3 +261,42 @@ bool sc_wait_for_file(const char *path, size_t timeout_sec)
 	}
 	return false;
 }
+
+const char *run_systemd_container = "/run/systemd/container";
+
+static bool _sc_is_in_container(const char *p)
+{
+	// see what systemd-detect-virt --container does in
+	// https://github.com/systemd/systemd/blob/5dcd6b1d55a1cfe247621d70f0e25d020de6e0ed/src/basic/virt.c#L749-L755
+	FILE *in SC_CLEANUP(sc_cleanup_file) = fopen(p, "r");
+	if (in == NULL) {
+		return false;
+	}
+
+	char container[128] = { 0 };
+
+	if (fgets(container, sizeof(container), in) == NULL) {
+		/* nothing read or other error? */
+		return false;
+	}
+
+	size_t r = strlen(container);
```

Comment:
I would like to see this as a helper, perhaps `sc_chomp` just to hide the string manipulation bits away from this function.

Reply:
@bboozzoo a static function in this file is fine.

Reply:
I'd to do it in a followup. I'll leave a TODO for now.